### PR TITLE
Space Hub Project

### PR DESCRIPTION
**Pull request by Leandro-Barretoo**
In this PR we made all the necessary requirements for Space Hub Project.

Its composed by 3 tabs:
- Rockets: Fetch and render list of available rockets plus functionality to do reservation of the same.
- Missions: Fetch and render list of available missions plus functionality to became a member.
- MyProfile: Show the active rockets and missions that a user has chosen.

Thank you for the review.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team 👋

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,2 +1,120 @@
...
 The Space Travelers' Hub consists of Rockets, Missions, and the My Profile section. The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking. The Missions section displays a list of current missions along with their brief description and participation status. The My Profile section displays all reserved rockets and space missions.
+
+This project is part of the Microverse curriculum module 3. The main objective is to understand how create a React app, use components, hooks, state, redux, run tests, and creating professional documentation for development.
+
+## Live Preview
+Heroku: coming soon!!!
+
+Netlify: coming soon!!!
+
+## Built With
```
Comment (mikenath223) [inline README.md]: - [ ] Kindly deploy the project online and include the live link in the readme.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+function App() {
+  return (
+    <>
+      <Navbar />
+      <Line />
+      <Routes>
+        <Route path="/rockets" element={<Rockets />} />
+        <Route path="/missions" element={<Missions />} />
+        <Route path="/my-profile" element={<MyProfile />} />
+        <Route path="/" element={<Rockets />} />
```
Comment (D4nielJ) [inline src/App.js:17]: [OPTIONAL] I think there is no need to repeat the rockets route here, it's fine if you leave it to be just the homepage, that way we avoid repetition.

File: src/api/api.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    const response = await fetch(url);
+    return response.json();
+  } catch (error) {
+    throw new Error(error.message);
+  }
+};
+
+export const fetchMissions = () => getData(missionsUrl);
+
+export const fetchRockets = () => getData(rocketsUrl);
```
Comment (D4nielJ) [inline src/api/api.js:15]: :clap:

File: src/components/Missions.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  const myMissionsArr = missionList.map((missions) => (
+    <MissionCard
+      key={missions.mission_id}
+      missionid={missions.mission_id}
+      name={missions.mission_name}
+      description={missions.description}
+      reserved={missions.reserved}
+    />
+  ));
```
Comment (D4nielJ) [inline src/components/Missions.js:41]: [OPTIONAL] We could make this block DRYer by destructuring `missions` when we map. Then we don't have to repeat `missions` each time

File: src/components/MyProfile.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          <tbody>
+            {renderReservedRockets}
+          </tbody>
+        </table>
+      </div>
+    </div>
+  );
+}
+
+export default MyProfile;
```
Comment (D4nielJ) [inline src/components/MyProfile.js:47]: [OPTIONAL] The My Profile section can be enhanced by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (no missions joined or no rockets reserved).

[OPTIONAL] The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.

[OPTIONAL] The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.

File: src/components/Rockets.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    <>
+      {
+        rockets.map((rocket) => (
+          <Rocket
+            key={rocket.id}
+            id={rocket.id}
+            image={rocket.image[0]}
+            name={rocket.name}
+            description={rocket.description}
+            reserved={rocket.reserved}
```
Comment (D4nielJ) [inline src/components/Rockets.js:27]: [OPTIONAL] Read my comment on destructuring. [You can read more about the Best practices of JS here. ](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) :slightly_smiling_face:

**Review by D4nielJ — APPROVED** [review]
## Hello @Leandro-Barretoo and @KDlamini,

### Great job!! Bravo!! :clap: :fire: :fireworks: :rocket: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**